src/source_distribution.rs

```diff
@@ -58,15 +58,16 @@ fn rewrite_cargo_toml(
                         dep_name
                     )
                 }
-                // This is the location of the targeted crate in the source distribution
-                table[&dep_name]["path"] = if root_crate {
-                    toml_edit::value(format!("{}/{}", LOCAL_DEPENDENCIES_FOLDER, dep_name))
-                } else {
-                    // Cargo.toml contains relative paths, and we're already in LOCAL_DEPENDENCIES_FOLDER
-                    toml_edit::value(format!("../{}", dep_name))
-                };
-                rewritten = true;
                 if !known_path_deps.contains_key(&dep_name) {
```

Comment:
Because of https://github.com/PyO3/maturin/pull/1084#issuecomment-1237686996, I think we need to somehow add the optional path deps to `known_path_deps` even if they're not activated via Cargo features.

Reply:
Unfortunately there isn't enough metadata:

```json
"dependencies": [
      {
          "name": "polars-algo",
          "source": null,
          "req": "^0.23.0",
          "kind": null,
          "rename": null,
          "optional": true,
          "uses_default_features": true,
          "features": [],
          "target": null,
          "registry": null,
          "path": "/Users/messense/Projects/polars/polars/polars-algo"
      }
]
```

and the optional `polars-algo` dep isn't present at `packages`.

Reply:
This reminds me of https://github.com/PyO3/maturin/pull/720#issuecomment-983423640, we might be able to workaround this by using `dependencies` instead of the structured `dep`.

https://github.com/koehlma/maturin/blob/dee86e60b1771055210d04fff5d8509d80a0fe04/src/source_distribution.rs#L146-L175